pkg/entities/ie_value.go

```diff
@@ -43,2317 +42,608 @@ type InfoElementWithValue interface {
-	return fmt.Errorf("setting value with wrong data type: %T", val)
+func (b *BaseInfoElement) SetSigned8Value(val int8) error {
+	panic("setting value with wrong data type")
 }
 
-func (s *StringInfoElement) SetSigned32Value(val int32) error {
-	return fmt.Errorf("setting value with wrong data type: %T", val)
+func (b *BaseInfoElement) SetSigned16Value(val int16) error {
+	panic("setting value with wrong data type")
 }
 
-func (s *StringInfoElement) SetSigned64Value(val int64) error {
-	return fmt.Errorf("setting value with wrong data type: %T", val)
+func (b *BaseInfoElement) SetSigned32Value(val int32) error {
+	panic("setting value with wrong data type")
 }
 
-func (s *StringInfoElement) SetFloat32Value(val float32) error {
-	return fmt.Errorf("setting value with wrong data type: %T", val)
+func (b *BaseInfoElement) SetSigned64Value(val int64) error {
+	panic("setting value with wrong data type")
 }
 
-func (s *StringInfoElement) SetFloat64Value(val float64) error {
-	return fmt.Errorf("setting value with wrong data type: %T", val)
+func (b *BaseInfoElement) SetFloat32Value(val float32) error {
+	panic("setting value with wrong data type")
 }
 
-func (s *StringInfoElement) SetBooleanValue(val bool) error {
-	return fmt.Errorf("setting value with wrong data type: %T", val)
+func (b *BaseInfoElement) SetFloat64Value(val float64) error {
+	panic("setting value with wrong data type")
 }
 
-func (s *StringInfoElement) SetMacAddressValue(val net.HardwareAddr) error {
-	return fmt.Errorf("setting value with wrong data type: %T", val)
+func (b *BaseInfoElement) SetBooleanValue(val bool) error {
+	panic("setting value with wrong data type")
 }
 
-func (s *StringInfoElement) SetStringValue(val string) error {
-	s.value = val
-	return nil
+func (b *BaseInfoElement) SetMacAddressValue(val net.HardwareAddr) error {
+	panic("setting value with wrong data type")
 }
 
-func (s *StringInfoElement) SetIPAddressValue(val net.IP) error {
-	return fmt.Errorf("setting value with wrong data type: %T", val)
+func (b *BaseInfoElement) SetStringValue(val string) error {
+	panic("setting value with wrong data type")
 }
 
-func (s *StringInfoElement) IsValueEmpty() bool {
-	return s.value == ""
+func (b *BaseInfoElement) SetIPAddressValue(val net.IP) error {
+	panic("setting value with wrong data type")
 }
 
-func (s *StringInfoElement) ResetValue() {
-	s.value = ""
+func (b *BaseInfoElement) GetLength() int {
+	return int(b.element.Len)
 }
 
-type DateTimeSecondsInfoElement struct {
-	value   uint32
-	element *InfoElement
+type Unsigned8InfoElement struct {
+	value uint8
+	BaseInfoElement
 }
 
-func NewDateTimeSecondsInfoElement(element *InfoElement, val uint32) *DateTimeSecondsInfoElement {
-	return &DateTimeSecondsInfoElement{
-		element: element,
-		value:   val,
+func NewUnsigned8InfoElement(element *InfoElement, val uint8) *Unsigned8InfoElement {
+	infoElem := &Unsigned8InfoElement{
+		value: val,
 	}
+	infoElem.element = element
+	return infoElem
```

Comment:
How about writing it in this way:
```
return &Unsigned8InfoElement{
	value: val,
        element: element,
}
```

Reply:
Embedded element cannot be accessed like that.